tests/test_timestamp.py

```diff
@@ -18,121 +18,147 @@
 import pytest
 
 from amazon.ion.core import Timestamp, TimestampPrecision, record
-from tests import parametrize
+from tests import parametrize, listify
 
 
-class _P(record('timestamps', 'expected_microseconds')):
+class _P(record('timestamps', 'expected_value')):
     def __str__(self):
         return self.desc
 
 
-MISSING_MICROSECOND = _P(
-    timestamps=[Timestamp(
-            2011, 1, 1,
-            0, 0, 0, None,
-            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=0
-            ),
-            Timestamp(
-            2011, 1, 1,
-            0, 0, 0, None,
-            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456')
-            ),
-            Timestamp(
-            2011, 1, 1,
-            0, 0, 0, None,
-            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123456789')
-            ),
-            Timestamp(
-            2011, 1, 1,
-            0, 0, 0, None,
-            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.000123')
-            ),
-            Timestamp(
-            2011, 1, 1,
-            0, 0, 0, None,
-            precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=Decimal('0.123000')
-            )],
-    expected_microseconds=[0, 123456, 123456, 123, 123000]
-)
+MISSING_MICROSECOND = [
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_precision=6, fractional_seconds=0
+        ), 0),
```

Comment:
So you are using a raw tuple here, but you have the `_P` record class as a named tuple above that you don't appear to be using.  The raw tuples are really hard to read, as a maintainer, what am I supposed to make of index `1` for example?

Why not make a `_P` class that has the `timestamp` and whatever field(s) your testing.  You could make that a dictionary to drive how what the tests are looking for as a way to be more data driven.

Reply:
What did you mean by this? I am using the `_P` class, which has `timestamps` and `expected_value` fields. I am populating these fields using the `event_type_parameters` method.

Reply:
Reading more carefully, I see what you're doing (and I realize that there are other code paths in the tests doing something similar).